views/helpers/annotations.js

```diff
@@ -0,0 +1,157 @@
+const extent = require("../helpers/extent.js");
+
+/**
+ * Returns true, if there is at least one annotation on the left or on the right.
+ */
+function hasAnnotationOnLeftOrRight(annotations, cssModifier) {
+  if (cssModifier === "narrow") return false; // on mobile...
+  return annotations.some(a => a.position === "left" || a.position === "right");
+}
+
+/**
+ * Returns true, if there is at least one annotation on the top or on the bottom.
+ */
+function hasAnnotationOnTopOrBottom(annotations, cssModifier) {
+  if (cssModifier === "narrow" && annotations.length > 0) return true; // on mobile...
+  return annotations.some(a => a.position === "top" || a.position === "bottom");
+}
+
+/**
+ * Returns true, if the region (Kanton, Landkreis, etc.) has at least one annotation.
+ */
+function regionHasAnnotation(annotations, region) {
+  return annotations.some(a => a.region === region);
+}
+
+/**
+ * Links the annotations to the hexagons and sets all the coordinates needed for drawing the annotations correctly
+ */
+function setCoordinatesForHexMap(annotations, hexagons, annotationStartPosition, cssModifier) {
+  const [xMin, xMax] = extent.getExtents(hexagons, ({ x }) => x);
+  const [yMin, yMax] = extent.getExtents(hexagons, ({ y }) => y);
+
+  annotations.forEach(a => {
```

Comment:
In this case I think it would be good to have a more expressive variable name, since it gets used quite a lot in the loop.